pkg/api/validation/path/name.go

```diff
@@ -35,23 +35,31 @@ func IsValidPathSegmentName(name string) []string {
 		}
 	}
 
+	message := []string{}
```

Comment:
declaring a slice var starts it out as nil, avoids the allocation in error-free cases, and removes the need to check for length:
```
var errors []string
...
for ... {
  errors = append(errors, ...)
}
return errors
```